### DISCUSSION TITLE
mapping du champ fulltext de l'entité Film

### DISCUSSION THREAD

**Discussion by Greywind299792458** [amt-classroom/amt-classroom.github.io#5] [Q&A] [answered]
Bonjour @bchapuis , 
nous hésitions sur la façon de mapper l'attribut fulltext de la table Film.

nous avons remarqué qu'il s'agissait d'un attribut de type "tsvector" et qu'à l'intérieur il s'agit d'une suite de mots clés associé à des nombres comme ci-dessous:
``` ... 'boat':21 'disturb':2 'fight':14 'frisbe':16 'lumberjack':11 'must':13 'stamped':1 'tale':5 ... ```
utilisée probablement pour de la recherche de mots clés parmi les films de la base ou quelque chose du genre.

Il nous a semblé qu'il n'existait pas d'équivalent direct pour mapper facilement cet attribut, et nous avions vu plusieurs pistes:
- nous sommes tombés sur cette pull request discutant du support du type tsvector qui semble avoir été acceptée mais qui impliquerait d'utiliser une librairie additionnelle: https://github.com/vladmihalcea/hypersistence-utils/issues/115
- nous avons également vu qu'il était possible de créer des type custom avec Hibernate et hésitions à utiliser cela ici

partons nous sur les bonnes pistes ou aurions nous raté quelque chose d'important ? Il nous semblait important de tout de même mapper cet attribut.

Cordialement,

Elliot et Alexis

**Comment by bchapuis** [comment, marked as answer]
Bonjour, comme mentionné en classe, cet attribtu est difficile à implémenter et nous proposons de simplement l'ignorer. Nous mettrons l'intitulé du laboratoire à jour de manière à clarifié ce point pour les futurs éditions du cours. C'est un type qui requiert l'implémentation d'un type (UserType) et vous devez déjà faire cet exercice pour le champs specialFeature de type text[].

**Reply by Greywind299792458** [reply]
ok, merci pour la clarification !
